graylog2-server/src/main/java/org/graylog/plugins/pipelineprocessor/rest/PipelineResource.java

```diff
@@ -250,6 +272,111 @@ public PipelineSource update(@ApiParam(name = "id") @PathParam("id") String id,
         return PipelineSource.fromDao(pipelineRuleParser, savedPipeline);
     }
 
+    public static record RoutingRequest(
+            @JsonProperty(value = "input_id", required = true) String inputId,
+            @JsonProperty(value = "stream_id", required = true) String streamId,
+            @JsonProperty(value = "remove_from_default", required = true) boolean removeFromDefault
+    ) {}
+
+    @ApiOperation(value = "Add a stream routing rule to the default routing pipeline.")
+    @Path("/routing")
+    @PUT
+    @AuditEvent(type = PipelineProcessorAuditEventTypes.PIPELINE_UPDATE)
+    public PipelineSource routing(@ApiParam(name = "body", required = true) @NotNull RoutingRequest request) throws NotFoundException {
+        checkPermission(RestPermissions.STREAMS_EDIT, request.streamId());
+        checkPermission(PipelineRestPermissions.PIPELINE_RULE_CREATE);
+
+        String streamName;
+        try {
+            final Stream stream = streamService.load(request.streamId());
+            streamName = stream.getTitle();
+        } catch (NotFoundException e) {
+            throw new InternalServerErrorException(f("Unable to load stream %s", request.streamId()), e);
+        }
+
+        RuleDao ruleDao = createRoutingRule(request, streamName);
+        PipelineDao pipelineDao;
+        try {
+            pipelineDao = pipelineService.loadByName(GL_INPUT_ROUTING_PIPELINE);
```

Comment:
It may also be worth checking whether the right pipeline connection (`PipelineConnections`) exists.